pkg/diff/sql_generator.go

```diff
@@ -1228,6 +1288,176 @@ func (a *attachPartitionSQLVertexGenerator) GetDeleteDependencies(_ schema.Table
+	}
+	sb.WriteString(fmt.Sprintf("\tSTART WITH %d CACHE %d %sCYCLE\n", seq.StartValue, seq.CacheSize, cycleModifier))
+
+	var hazards []MigrationHazard
+	if seq.Owner == nil {
+		hazards = append(hazards, migrationHazardSequenceCannotTrackDependencies)
+	}
+
+	return []Statement{{
+		DDL:     sb.String(),
+		Timeout: statementTimeoutDefault,
+		Hazards: hazards,
+	}}, nil
+}
+
+func (s *sequenceSQLVertexGenerator) Delete(seq schema.Sequence) ([]Statement, error) {
+	var hazards []MigrationHazard
+	if seq.Owner != nil && (s.isDeletedWithOwningTable(seq) || s.isDeletedWithColumns(seq)) {
+		return nil, nil
+	} else if seq.Owner == nil {
+		hazards = append(hazards, migrationHazardSequenceCannotTrackDependencies)
+	}
+	return []Statement{{
+		DDL:     fmt.Sprintf("DROP SEQUENCE %s", seq.GetFQEscapedName()),
+		Timeout: statementTimeoutDefault,
+		Hazards: hazards,
+	}}, nil
+}
+
+func (s *sequenceSQLVertexGenerator) Alter(diff sequenceDiff) ([]Statement, error) {
+	// Ownership changes handled by the sequenceOwnershipSQLVertexGenerator
+	diff.old.Owner = diff.new.Owner
+	if !cmp.Equal(diff.old, diff.new) {
+		// Technically, we could support altering expression, but I don't see the use case for it. it would require more test
+		// cases than forceReadding it, and I'm not convinced it unlocks any functionality
+		return nil, fmt.Errorf("altering check constraint to resolve the following diff %s: %w", cmp.Diff(diff.old, diff.new), ErrNotImplemented)
+	}
+
+	return nil, nil
+}
+
+func (s *sequenceSQLVertexGenerator) GetSQLVertexId(seq schema.Sequence) string {
+	return buildSequenceVertexId(seq.SchemaQualifiedName)
+}
+
+func (s *sequenceSQLVertexGenerator) GetAddAlterDependencies(new schema.Sequence, _ schema.Sequence) []dependency {
+	deps := []dependency{
+		mustRun(s.GetSQLVertexId(new), diffTypeAddAlter).after(s.GetSQLVertexId(new), diffTypeDelete),
+	}
+	if new.Owner != nil {
+		// Sequences should be added/altered before the table they are owned by
+		deps = append(deps, mustRun(s.GetSQLVertexId(new), diffTypeAddAlter).before(buildTableVertexId(new.Owner.TableUnescapedName), diffTypeAddAlter))
+	}
+	return deps
+}
+
+func (s *sequenceSQLVertexGenerator) GetDeleteDependencies(seq schema.Sequence) []dependency {
+	var deps []dependency
+	// This is an unfortunate hackaround. It would make sense to also have a dependency on the owner column, such that
+	// the sequence can only be considered deleted after the owning column is deleted. However, we currently don't separate
+	// column deletes from table add/alters. We can't build this dependency without possibly creating a circular dependency:
+	// the sequence add/alter must occur before the new owner column add/alter, but the sequence delete must occur after the
+	// old owner column delete (equivalent to add/alter) and the sequence add/alter. We can get away with this because
+	// we, so far, no columns are ever "re-created". If we ever do support that, we'll need to revisit this.
```

Comment:
```suggestion
	// old owner column delete (equivalent to add/alter) and the sequence add/alter. We can get away with this because
	// so far, no columns are ever "re-created". If we ever do support that, we'll need to revisit this.
```